metagraph/src/graph/representation/succinct/dbg_succinct.cpp

```diff
@@ -770,24 +770,29 @@ void DBGSuccinct::serialize(const std::string &filename) const {
             throw std::ios_base::failure("Can't write to file " + out_filename);
     }
 
-    if (!valid_edges_.get())
-        return;
+    auto serialize_valid_edges = [&](auto &&valid_edges) {
+        assert((boss_graph_->get_state() == BOSS::State::STAT
+                    && dynamic_cast<const bit_vector_small*>(valid_edges.get()))
+            || (boss_graph_->get_state() == BOSS::State::FAST
+                    && dynamic_cast<const bit_vector_stat*>(valid_edges.get()))
+            || (boss_graph_->get_state() == BOSS::State::DYN
+                    && dynamic_cast<const bit_vector_dyn*>(valid_edges.get()))
+            || (boss_graph_->get_state() == BOSS::State::SMALL
+                    && dynamic_cast<const bit_vector_small*>(valid_edges.get())));
+
+        const auto out_filename = prefix + kDummyMaskExtension;
+        std::ofstream out = utils::open_new_ofstream(out_filename);
+        if (!out.good())
+            throw std::ios_base::failure("Can't write to file " + out_filename);
 
-    assert((boss_graph_->get_state() == BOSS::State::STAT
-                && dynamic_cast<const bit_vector_small*>(valid_edges_.get()))
-        || (boss_graph_->get_state() == BOSS::State::FAST
-                && dynamic_cast<const bit_vector_stat*>(valid_edges_.get()))
-        || (boss_graph_->get_state() == BOSS::State::DYN
-                && dynamic_cast<const bit_vector_dyn*>(valid_edges_.get()))
-        || (boss_graph_->get_state() == BOSS::State::SMALL
-                && dynamic_cast<const bit_vector_small*>(valid_edges_.get())));
-
-    const auto out_filename = prefix + kDummyMaskExtension;
-    std::ofstream out = utils::open_new_ofstream(out_filename);
-    if (!out.good())
-        throw std::ios_base::failure("Can't write to file " + out_filename);
+        valid_edges->serialize(out);
+    };
 
-    valid_edges_->serialize(out);
+    if (valid_edges_) {
+        serialize_valid_edges(valid_edges_);
+    } else {
+        serialize_valid_edges(generate_dummy_kmers(1, false));
```

Comment:
With the new indexing, I _really_ don't want to allow these graphs to walk around with invalid `in_graph` function. Would lead (and leads if this is removed) to all kinds of hard to debug behaviors. But I'm also not sure what's the best place to propagate this information, and how much overhead it may add?